### PR TITLE
Remove BaseModel since it breaks equals/hashCode contract

### PR DESCRIPTION
**Pull request by Weltraumschaf**
Problem of equals and inheritance:
As described in this[1] blog post the Object#equals() requires that it fulfills the Liskov Substitution Principle (LSP). Our implemenation breaks this contract. Even worse, it didn't work at all as descibed in issue 23[2].

Since the BaseModel class does not have any properties, we can remove it.

This patch removes this obsolete class. Also it makes all model classes final and all properties private. This is an implicit requirement of the contract of hashCode() to avoid memory leaks in collections. (See linked blog post on Artima for more details.)

Actually objects must be immutable -- all fields final -- to guaruantee a stable equal/hashCode behaviour for the whole lifetime of the objects. But this must be further investigated, if this is possible. For now we ignore this warning in the test code.

1: https://www.artima.com/articles/how-to-write-an-equality-method-in-java
2: https://github.com/secureCodeBox/defectdojo-client-java/issues/23

### CODE REVIEW

**Comment by Weltraumschaf** [comment]
I suggest reviewing by commits instead of files bc I did a reformatting of all classes in one commit.

**Review by J12934 — DISMISSED** [review]
looks generally good added a couple of comments here to todo question in case that helps give some context

**Comment by J12934** [comment]
which commits here are new since the last review?
don't want to go through all of them again and with the force pushes & rebases on this branch the dates are wrong and the github ui doesn't show me where to start again.

File: src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+   * This is {@code null} if there is no next page.
+   * </p>
+   */
+  @JsonProperty
+  private String next;
+
+  /**
+   * TODO: What does this contain? I would expect a number for the previous page.
+   * <p>
+   * This is {@code null} if there is no next page.
```
Comment (J12934) [inline src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java:38]:
```suggestion
   * This is {@code null} if there is no previous page.
```

File: src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * This class wraps the paginated results from DefectDojo
+ *
+ * @param <T> type of results
+ */
+@Data
+public final class PaginatedResult<T extends Model> {
+  /**
+   * TODO: What does this count? The number of results in one page or the total number?
```
Comment (J12934) [inline src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java:21]: total

File: src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   */
+  @JsonProperty
+  private int count;
+
+  /**
+   * TODO: What does this contain? I would expect a number for the next page.
```
Comment (J12934) [inline src/main/java/io/securecodebox/persistence/defectdojo/model/PaginatedResult.java:27]: if i remember right this is the url for a get request to the next page

File: src/main/java/io/securecodebox/persistence/defectdojo/model/ScanFile.java [hunk 4]
```diff
@@ -6,28 +6,30 @@
 
 import lombok.Data;
 
+// TODO: Why we do not have as many annotations as the other models here?
+// TODO: Why does this class does not implement Model?
```
Comment (J12934) [inline src/main/java/io/securecodebox/persistence/defectdojo/model/ScanFile.java:10]: if i remember right this is the class used for the multi part form upload, so its not a json model

File: src/main/java/io/securecodebox/persistence/defectdojo/model/UserProfile.java [hunk 5]
```diff
@@ -23,7 +23,10 @@ public final class UserProfile implements Model {
 
   @Override
   public boolean equalsQueryString(Map<String, Object> queryParams) {
-    // The user_profile endpoint does not have query parameters that's why this function will just return true
+    // The user_profile endpoint does not have query parameters that's
+    // why this function will just return true.
+    // TODO: All other implementations return false if null is given for queryParams.
```
[the remark this answers is not in the document]
Comment (J12934) [reply src/main/java/io/securecodebox/persistence/defectdojo/model/UserProfile.java:28]: you resolved this but the todo is still here?

[the remark this answers is not in the document]
Comment (Weltraumschaf) [reply src/main/java/io/securecodebox/persistence/defectdojo/model/UserProfile.java]: TODO is done.

**Review by J12934 — APPROVED** [review]

**Pull request merged**